### PR TITLE
Mingw: Replace some more _MSC_VER with _WIN32

### PR DESCRIPTION
**Pull request by 1480c1**

### CODE REVIEW

**Comment by tianjunwork** [comment]
Thanks for the patch 👍
Pls add sign-off line.

**Comment by 1480c1** [comment]
Will do, just want to see if I can test with generic mingw-w64 toolchain instead of one provided by msys2, but having a hard time finding a good/official one that's up to date

File: Source/Lib/Codec/EbDefinitions.h [hunk 1]
```diff
@@ -104,10 +104,10 @@ Groups of Pictures (GOP) units.
...
 #define EB_PRED_TOTAL_COUNT     3
 #define EB_PRED_INVALID         0xFF
-    
+
 // *Note - This work around is needed for the windows visual studio compiler
-//  (MSVC) because it doesn't support the C99 header file stdint.h.  
-//  All other compilers should support the stdint.h C99 standard types. 
+//  (MSVC) because it doesn't support the C99 header file stdint.h.
+//  All other compilers should support the stdint.h C99 standard types.
 #ifdef _MSC_VER
```
Comment (tianjunwork) [inline Source/Lib/Codec/EbDefinitions.h]: one more _MSC_VER?
I recommend do: grep -wr _MSC_VER once again.

File: Source/Lib/Codec/EbUtility.c [hunk 2]
```diff
@@ -3,26 +3,14 @@
...
-#include "EbDefinitions.h"
-//#if  (WIN_ENCODER_TIMING || WIN_DECODER_TIMING)
 #include <time.h>
-//#endif
-#elif __linux__
-#include <stdio.h>
-#include <stdlib.h>
-#include "EbDefinitions.h"
+#ifndef _WIN32
 //#if   (LINUX_ENCODER_TIMING || LINUX_DECODER_TIMING)
```
Comment (tianjunwork) [inline Source/Lib/Codec/EbUtility.c]: also remove //#if   (LINUX_ENCODER_TIMING || LINUX_DECODER_TIMING) and //#endif?

Comment (1480c1) [reply Source/Lib/Codec/EbUtility.c]: Sure, wasn't sure if this should've been removed or not

File: Source/Lib/Codec/EbDefinitions.h (hunk 1 above)
Comment (1480c1) [reply Source/Lib/Codec/EbDefinitions.h]: Some things are msvc only, mingw-w64 is based on gcc

**Comment by Jamaika1** [comment]
In file included from c:\msys911\lib\gcc\x86_64-w64-mingw32\9.1.1\include\immintrin.h:43,
                 from EbDefinitions.h:234,
                 from EbEncHandle.c:17:
EbEncHandle.c: In function 'CheckXcr0Ymm':
c:\msys911\lib\gcc\x86_64-w64-mingw32\9.1.1\include\xsaveintrin.h:60:1: error: inlining failed in call to always_inline '_xgetbv': target specific option mismatch
   60 | _xgetbv (unsigned int __A)
      | ^~~~~~~
EbEncHandle.c:516:22: note: called from here
  516 |     xcr0 = (uint32_t)_xgetbv(0);  /* min VS2010 SP1 compiler is required */

**Comment by 1480c1** [comment]
Can you provide your cmake commands/other commands and how did you get your mingw-w64 toolchain?

**Comment by Jamaika1** [comment]
for %%f in ("EbEncHandle.c") do gcc.exe -DNDEBUG -fPIC -ftree-vectorize -g0 -O3 -DWIN32 -DNNON_AVX512_SUPPORT -c %%f -o %%~nf.o

**Comment by 1480c1** [comment]
Why would you compile the files themselves?

**Comment by Jamaika1** [comment]
Previously worked. Now it's different.

**Comment by 1480c1** [comment]
Do you not wish to use cmake?


Either way, it requires `-mxsave` for gcc

**Comment by Jamaika1** [comment]
OK, but I would like to have an instruction for SSE2 and not AVX.

**Comment by 1480c1** [comment]
What kind of instruction?

**Pull request merged**